modules/openapi-generator/src/main/resources/python/python-experimental/signing.mustache

```diff
@@ -0,0 +1,317 @@
+            raise Exception("The signature max validity must be a positive value")
+        self.signature_max_validity = signature_max_validity
+        """The signature max validity, expressed as a datetime.timedelta value.
+        It must be a positive value.
+        """
+        # If the user has not provided any signed_headers, the default must be set to '(created)'.
+        if signed_headers is None or len(signed_headers) == 0:
+            signed_headers = [HEADER_CREATED]
+        if self.signature_max_validity is None and HEADER_EXPIRES in signed_headers:
+            raise Exception(
+                "Signature max validity must be set when " \
+                "'(expires)' signature parameter is specified")
+        if len(signed_headers) != len(set(signed_headers)):
+            raise Exception("Cannot have duplicates in the signed_headers parameter")
+        if HEADER_AUTHORIZATION in signed_headers:
+            raise Exception("'Authorization' header cannot be included in signed headers")
+        self.signed_headers = signed_headers
+        """A list of strings. Each value is the name of HTTP header that must be included
+           in the HTTP signature calculation.
+        """
+        self.private_key = None
+        """The private key used to sign HTTP requests.
+            Initialized when the PEM-encoded private key is loaded from a file.
+        """
+
+    def get_http_signature_headers(self, host, resource_path, method, headers, body, query_params):
+        """
+        Create a cryptographic message signature for the HTTP request and add the signed headers.
+        
+        :param resource_path : A string representation of the HTTP request resource path.
+        :param method: A string representation of the HTTP request method, e.g. GET, POST.
+        :param headers: A dict containing the HTTP request headers.
+        :param body: The string representation of the HTTP request body.
+        :param query_params: A string representing the HTTP request query parameters.
+        :return: A dict of HTTP headers that must be added to the outbound HTTP request.
+        """
+        if method is None:
+            raise Exception("HTTP method must be set")
+        if resource_path is None:
+            raise Exception("Resource path must be set")
+
+        signed_headers_dict, request_headers_dict = self.get_signed_header_info(
+            host, resource_path, method, headers, body, query_params)
+
+        header_items = [
+            "{0}: {1}".format(key.lower(), value) for key, value in signed_headers_dict.items()]
+        string_to_sign = "\n".join(header_items)
+
+        digest, digest_prefix = self.get_message_digest(string_to_sign.encode())
+        b64_signed_msg = self.sign_digest(digest)
+
+        request_headers_dict[HEADER_AUTHORIZATION] = self.get_authorization_header(
+            signed_headers_dict, b64_signed_msg)
+
+        return request_headers_dict
+
+    def load_private_key(self):
```

Comment:
Can change this method's name to add a `__` prefix so we can tell that it is not our publicly used method?
See http://python.robasworld.com/private-member-name-mangling/

Reply:
done